In [ ]:
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier 
from sklearn.svm import SVC
from sklearn.datasets import load_iris
from sklearn.datasets import load_files

In [ ]:
iris = load_iris()

X = iris.data
y = iris.target

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.4, random_state=1)

dt_clf = DecisionTreeClassifier()
svc_clf = SVC()
knn_clf = KNeighborsClassifier()

klasyfikatory = [dt_clf, svc_clf, knn_clf]

for clf in klasyfikatory:
    print("--------------")
    print("fitting - training...")
    clf.fit(X_train, y_train)

    print("training on whole dataset...")
    clf.fit(X, y)

    print("predicting...")
    y_pred = clf.predict(X_test)

    print("true values ", y[:10])
    print("predicted   ", y_pred[:10])

    print("scoring...")

    clf_score = clf.score(X_train, y_train)
    print("Train score = ", clf_score)

    clf_score = clf.score(X_test, y_test)
    print("Test score = ", clf_score)

    clf_score = clf.score(X, y)
    print("whole set score = ", clf_score)

--------------
fitting - training...
training on whole dataset...
predicting...
true values  [0 0 0 0 0 0 0 0 0 0]
predicted    [0 1 1 0 2 1 2 0 0 2]
scoring...
Train score =  1.0
Test score =  1.0
whole set score =  1.0
--------------
fitting - training...
training on whole dataset...
predicting...
true values  [0 0 0 0 0 0 0 0 0 0]
predicted    [0 1 1 0 2 1 2 0 0 2]
scoring...
Train score =  0.9666666666666667
Test score =  0.9833333333333333
whole set score =  0.9733333333333334
--------------
fitting - training...
training on whole dataset...
predicting...
true values  [0 0 0 0 0 0 0 0 0 0]
predicted    [0 1 1 0 2 1 2 0 0 2]
scoring...
Train score =  0.9555555555555556
Test score =  0.9833333333333333
whole set score =  0.9666666666666667


1. Skorzystałem z dataset "iris". Dane zostały podzielone w stosunku 80% do 20%.

2. Użyłem 3 klasyfikatory:

*   DecisionTreeClassifier
*   SVC
*   KNeighborsClassifier

3. Wytrenowałem oraz obliczyłem metryki dla każdego klasyfikatora (klasyfikator - train score - test score):

*   DecisionTreeClassifier - 1.000 
*   SVC - 0.966 - 0.983
*   KNeighborsClassifier - 0.955 - 0.983

SVC, KNeighborsClassifier stworzyły bardzo dobre modele, których metryki na danych testowych wypadły niewiele słabiej niż na danych treningowych. Wśród nich najlepsze okazało się SVC.

DecisionTreeClassifier - mamy tutaj do czynienia z overfittingiem. Dla obu rodzajów danych mamy wartość = 1